### ISSUE TITLE
Update removed all filters

### ISSUE THREAD

**Issue by mshale87** [afterxleep/Bouncer#49]
**Describe the bug**
iPhone App Store auto updated Bouncer to 2.3.0 and then over night all of my filters were gone and I have hundreds of unread text messages that are spam- I have since deleted them. There no categories listed from Bouncer in Messages app, and there are no filters present in Bouncer anymore. Settings showed that Bouncer became unselected. 

**To Reproduce**

**Expected behavior**

**Screenshots**
![8DD8F626-39A8-4EA5-BEE4-697FCBCCAB41](https://user-images.githubusercontent.com/65051424/235765022-58b5d9fe-9b11-4b41-9c6f-314ec6b3c422.png)
![5396087C-6C3B-41B0-AD3E-8C6E61E1A7B2](https://user-images.githubusercontent.com/65051424/235765025-867f9e2d-a9ef-4b7e-b460-f9a2e7e12672.png)
![AD4A81A4-B12A-4B38-8535-DF1115DD9A12](https://user-images.githubusercontent.com/65051424/235765026-6fdf4d27-7718-40dc-834e-c438ae717027.png)

**Desktop (please complete the following information):**
 - iOS Version: 16.3.1
 - Device: iPhone 12 Pro

**Additional context**

**Comment by afterxleep** [comment]
Hi @mshale87.  

An error could've caused this during the migration.  Do you happen to know which version you had before?

**Comment by mshale87** [comment]
I have the phone set up auto update apps. So I likely had 2.2.3 since it was released about 7 months ago. Hope that helps!- Matt HaleOn May 2, 2023, at 4:03 PM, Daniel Bernal ***@***.***> wrote:﻿
Hi @mshale87.
An error could've caused this during the migration.  Do you happen to know which version you had before?

**Comment by afterxleep** [comment]
Thank you @mshale87.

I did check the migration script for the current version and did not find any issues.  I'm sorry about the data loss, but not really sure what could've happened.

Do you see iMessage categories when selecting Bouncer as you filter?

Let me know,

Daniel

**Comment by mshale87** [comment]
Yes, after reselecting Bouncer in Settings-Messages-filter, I can see a lot of default filter categories that were not there previously. However, the filter list in Bouncer is still very empty. I suppose I will just have to rebuild the filters I made long ago! Will the filters apply to current messages or only new incoming messages?- Matt HaleOn May 3, 2023, at 4:01 AM, Daniel Bernal ***@***.***> wrote:﻿
Thank you @mshale87.
I did check the migration script for the current version and did not find any issues.  I'm sorry about the data loss, but not really sure what could've happened.
Do you see iMessage categories when selecting Bouncer as you filter?
Let me know,
Daniel

**Comment by afterxleep** [comment]
Yea, unfortunately it seems you'll have to re-create the filters, but Apple does not allow applying filters to already delivered messages, so likely you'll have to delete them by hand.   Sorry about the troble.  I'll follow up with a review to try understand what happened there.

**Comment by leres** [comment]
This happened to me as well.

Even though no rules were shown I tried "export block list" and was able to email myself a json blob with 86 rules. I was also able to save it to files but when I try to import I get:

The file "filters.json" couldn't be opened because you don't have permission to view it.

I looked around but couldn't immediately find documentation on how to use import. (When I select  "help" from the app pulldown nothing happens.)

I'd rather not post my filters but am happy  to share them off-line. But here's an example:
```
    {
        "id": "647938C1-4A88-4E93-B1C7-165DD3D1A723",
        "phrase": "\\.run\\b",
        "useRegex": true,
        "type": "message",
        "action": "junk"
    },
```

**Comment by afterxleep** [comment]
Hi @leres 

Thanks for the details.  Yes that import error is a bug that was reported to me yesterday, caused by some permissions being different in the latest iOS.  

I've submitted a new version to Apple and is waiting for approval.  In the meantime, you should be able to get 2.3.1 beta soon from the public testing group as beta approvals are a lot faster.

I'm not sure you would be able to import your filters directly from that list without updating to the new format, but is worth a shot.  If that does not work, I can help with info about the new format, or give you a hand with updating the json so it works.

**Comment by afterxleep** [comment]
@leres 

If you are willing to share your filters privately, theres "Need Help" link in the help section of the app that you can use to send them directly to me.   That would help me understand the issue a bit better.

**Comment by leres** [comment]
Sounds like first I need to wait for the new version to hit the apple store before  I'll even be able to try and import anything?

I think I can create a couple of rules and export to see what the new json format looks like and then know how to update my old rules.

As I said before, when I select "help" from the app pulldown nothing happens so I don't even see the "Need Help" link.

**Comment by leres** [comment]
Ok, so I installed the testflight version on my backup phone. First I tried importing my old format json file, that gives INCORRECT_FILE_FORMAT.

Then I created a filter, exported, and saw that my old format was missing `caseSensitive` and `subAction`; I added those and still get INCORRECT_FILE_FORMAT.

I emailed you my modified filter list, I'm sure it'll be obvious what the problem is.

**Comment by leres** [comment]
I have a M2 mini so I fired up xcode and after (painfully) uploading an updated json file to the simulator I get this unhelpful error:

`
2023-05-07 11:06:25.844936-0700 Bouncer[21056:1499222] [DocumentManager] The view service did terminate with error: Error Domain=_UIViewServiceErrorDomain Code=1 "(null)" UserInfo={Terminated=disconnect method}`

**Comment by afterxleep** [comment]
Hey @leres.  

That error, you can safely ignore, it has nothing to do with the import.

I’ve sent you back a fixed file, but that also helped me realize the original issue, which should be fixed in 2.3.2.

Some filters were missing the `useRegex` bool, which was optional in 2.2.x but required in 2.3.x.

Let me know if it works.

**Comment by leres** [comment]
Hah! I installed the testflight 2.3.2 version on my main phone and **bam**, all my filters are back! And I upgraded my backup phone and imported the filters so now I have them there as well.

Thanks for (the app and) the quick fixes!

**Comment by mshale87** [comment]
I can also confirm that the test flight version has all of my filters back with no export/import needed. - Matt HaleOn May 7, 2023, at 3:28 PM, Craig Leres ***@***.***> wrote:﻿
Hah! I installed the testflight 2.3.2 version on my main phone and bam, all my filters are back! And I upgraded my backup phone and imported the filters so now I have them there as well.
Thanks for (the app and) the quick fixes!
